### PR TITLE
keybase id using Identify2

### PR DESCRIPTION
**Pull request by patrickxb**
removed --track-statement flag
allowing resolve to work without uid, assertion if logged in

r? @maxtaco

### CODE REVIEW

File: go/engine/resolve_identify2.go [hunk 1]
```diff
@@ -49,6 +49,19 @@ func (e *ResolveThenIdentify2) resolveUID(ctx *Context) (err error) {
 		return nil
 	}
 
+	// if no uid and no assertion, then if logged in use self uid:
+	if len(e.arg.UserAssertion) == 0 {
```
Comment (maxtaco) [inline go/engine/resolve_identify2.go]: are we sure we always want this behavior?  It makes sense in the context of `keybase id` but I'm not sure if it does everywhere else?

**Comment by maxtaco** [comment]
is there any code we can delete in `service/identify` as a result of this fix?

File: go/client/cmd_id.go [hunk 2]
```diff
@@ -18,7 +18,6 @@ import (
 type CmdID struct {
 	libkb.Contextified
 	user           string
-	trackStatement bool
```
Comment (maxtaco) [inline go/client/cmd_id.go]: can we nuke the code that handled this on the backend?

File: go/engine/resolve_identify2.go (hunk 1 above)
Comment (maxtaco) [inline go/engine/resolve_identify2.go]: We might want to trigger this behavior with an explicit flag.

**Comment by maxtaco** [comment]
Done with 1st pass, might be worth another depending on your questions to the above!

**Comment by patrickxb** [comment]
the `Identify` handler, possibly, but I'm not sure what the status is of other clients using that endpoint.

File: go/client/cmd_id.go (hunk 2 above)
Comment (patrickxb) [inline go/client/cmd_id.go]: Just the track statement part?  Sure.

**Comment by maxtaco** [comment]
KBFS doesn't use it, so as long as `keybase/client` doesn't use it, it is OK to nuke.

**Comment by maxtaco** [comment]
Seems like some of the tests still use `Identify()`.  OK, leave in!

**Comment by patrickxb** [comment]
What about desktop app?

**Comment by maxtaco** [comment]
I don't think the desktop app (currently) ever calls identify.  It just implements the U/I from KBFS.

File: go/engine/resolve_identify2.go (hunk 1 above)
Comment (patrickxb) [inline go/engine/resolve_identify2.go]: Added.

**Comment by patrickxb** [comment]
PTAL @maxtaco

File: go/engine/resolve_identify2.go [hunk 3]
```diff
@@ -49,6 +49,19 @@ func (e *ResolveThenIdentify2) resolveUID(ctx *Context) (err error) {
...
+		if ok {
+			e.arg.Uid = uid
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		return libkb.LoginRequiredError{Context: "to identify without specifying a user assertion"}
+	}
+
```
Comment (maxtaco) [inline go/engine/resolve_identify2.go]: and what happens if we don't have a user assertion and `AllowEmptySelfID` is `false`?

**Comment by maxtaco** [comment]
Had a small question, but otherwise OK

File: go/engine/resolve_identify2.go (hunk 3 above)
Comment (patrickxb) [inline go/engine/resolve_identify2.go]: The same thing that happened before this change.  An error.

Comment (maxtaco) [inline go/engine/resolve_identify2.go]: ok

**Pull request merged**